packages/@react-spectrum/searchwithin/test/SearchWithin.test.js

```diff
@@ -152,20 +152,45 @@ describe('SearchWithin', function () {
 
     expect(searchfield).not.toHaveAttribute('aria-required');
 
-    expect(searchfield).toHaveAttribute('aria-labelledby', label.id);
+    expect(searchfield).toHaveAttribute('aria-labelledby', `${group.id} ${searchfield.id}`);
     expect(group).toHaveAttribute('aria-labelledby', label.id);
 
     expect(searchfield.classList.contains('is-quiet')).toBeFalsy();
     expect(picker.classList.contains('spectrum-Dropdown--quiet')).toBeFalsy();
   });
 
-  it('searchfield and group are labelled correctly if aria-label used', function () {
-    let {getByRole, getByLabelText} = renderSearchWithin({label: undefined, 'aria-label': 'Aria Label'});
+  it('Should handle aria-labels without visible label', function () {
+    let {getByRole, getByPlaceholderText} = renderSearchWithin({label: undefined, 'aria-label': 'Aria Label'});
 
-    let searchfield = getByLabelText('Aria Label', {selector: 'input'});
     let group = getByRole('group');
+    let searchfield = getByPlaceholderText('Search');
+    let picker = getByRole('button');
 
+    expect(group).toHaveAttribute('aria-label', 'Aria Label');
     expect(searchfield).toHaveAttribute('aria-label', 'Aria Label');
-    expect(group).toHaveAttribute('aria-labelledby', 'Aria Label');
+    expect(picker).toHaveAttribute('aria-label', 'Search within');
+
+    expect(group).not.toHaveAttribute('aria-labelledby');
+    expect(searchfield).toHaveAttribute('aria-labelledby', `${group.id} ${searchfield.id}`);
+    expect(picker).toHaveAttribute('aria-labelledby', `${group.id} ${picker.id} ${picker.childNodes[0].id}`);
+  });
+
+  it('Should use default aria-labels if no aria-label provided', function () {
+    let consoleWarnSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
+    let {getByRole, getByPlaceholderText} = renderSearchWithin({label: undefined});
+
+    let group = getByRole('group');
+    let searchfield = getByPlaceholderText('Search');
+    let picker = getByRole('button');
+
+    expect(group).toHaveAttribute('aria-label', 'Search');
+    expect(searchfield).toHaveAttribute('aria-label', 'Search');
+    expect(picker).toHaveAttribute('aria-label', 'Search within');
+
+    expect(group).not.toHaveAttribute('aria-labelledby');
+    expect(searchfield).toHaveAttribute('aria-labelledby');
+    expect(picker).toHaveAttribute('aria-labelledby', `${group.id} ${picker.id} ${picker.childNodes[0].id}`);
+
+    expect(consoleWarnSpy).toHaveBeenLastCalledWith('If you do not provide a visible label, you must specify an aria-label or aria-labelledby attribute for accessibility');
```

Comment:
i'm a bit confused, why is this being called? it looks like everything has a default label?

Reply:
This happens because useLabel doesn't have a default label when it's called, and I don't think we want this. I think this can be fixed by doing something like this:
```jsx
let labels = useLabels(props, formatMessage('search'));
let {labelProps, fieldProps} = useLabel({
  ...props,
  ...labels
});
```
Thanks! Pushing a fix.